ocaml/testsuite/tests/typing-modes/val_modalities.ml

```diff
@@ -463,3 +463,89 @@ Error: Signature mismatch:
          val f : int -> int @@ portable
        The second is portable and the first is not.
 |}]
+
+(* Including module type with modalities *)
+module type S = sig
+  val foo : 'a -> 'a
+
+  val bar : 'a -> 'a @@ nonportable
```

Comment:
Being able to write `@@nonportable` here is confusing. Let's fix this in another PR.